spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java

```diff
@@ -166,7 +166,9 @@ public BinderAwareChannelResolver binderAwareChannelResolver(BindingService bind
 	@Bean
 	@ConditionalOnProperty("spring.cloud.stream.bindings." + ERROR_CHANNEL_NAME + ".destination")
 	public SingleBindingTargetBindable<MessageChannel> errorChannelBindable(
-			@Qualifier(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME) PublishSubscribeChannel errorChannel) {
+			@Qualifier(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME) PublishSubscribeChannel errorChannel,
+			CompositeMessageChannelConfigurer compositeMessageChannelConfigurer) {
+		compositeMessageChannelConfigurer.configureOutputChannel(errorChannel, ERROR_CHANNEL_NAME);
```

Comment:
Instead of directly manipulating the error channel, it would be better to bridge it to a secondary channel and actually bind the latter. Applying the interceptors directly on the error channel means that all its recipients will start receiving marshaled objects, and so on.